gateway/api/runbooks/runbooksV2.go

```diff
@@ -0,0 +1,434 @@
+//	@Tags			Runbooks
+//	@Produce		json
+//	@Success		200			{object}	openapi.RunbookList
+//	@Failure		404,422,500	{object}	openapi.HTTPError
+//	@Router			/runbooks [get]
+func ListRunbooksV2(c *gin.Context) {
+	ctx := storagev2.ParseContext(c)
+
+	runbookConfig, err := models.GetRunbookConfigurationByOrgID(models.DB, ctx.GetOrgID())
+	if err != nil {
+		log.Infof("failed fetching runbook configuration, err=%v", err)
+		c.JSON(http.StatusUnprocessableEntity, gin.H{"message": "failed fetching runbook configuration"})
+		return
+	}
+
+	runbookRules, err := models.GetRunbookRules(models.DB, ctx.OrgID, 0, 0)
+	if err != nil {
+		log.Infof("failed fetching runbook rules, err=%v", err)
+		c.JSON(http.StatusUnprocessableEntity, gin.H{"message": "failed fetching runbook rules"})
+		return
+	}
+
+	urlQuery := c.Request.URL.Query()
+	connection := urlQuery.Get("connection")
+
+	removeEmptyConnectionsList := false
+	var connectionNames []string
+	if connection != "" {
+		connectionNames = []string{connection}
+		removeEmptyConnectionsList = true
+	} else {
+		connectionNames, err = models.ListConnectionsName(models.DB, ctx.GetOrgID())
+		if err != nil {
+			log.Infof("failed fetching connection names, err=%v", err)
+			c.JSON(http.StatusUnprocessableEntity, gin.H{"message": "failed fetching connection names"})
+			return
+		}
+	}
+
+	runbookList := &openapi.RunbookListV2{
+		Repositories: []openapi.RunbookRepositoryList{},
+	}
+	for _, configEnvVars := range runbookConfig.RepositoryConfigs {
+		config, err := runbooks.NewConfigV2(configEnvVars)
+		if err != nil {
+			c.JSON(http.StatusUnprocessableEntity, gin.H{"message": err.Error()})
+			return
+		}
+		repositoryList, err := listRunbookFilesV2(ctx.OrgID, config, runbookRules, connectionNames, ctx.UserGroups, removeEmptyConnectionsList)
+		if err != nil {
+			log.Infof("failed listing runbooks, err=%v", err)
+			c.JSON(http.StatusUnprocessableEntity, gin.H{"message": fmt.Sprintf("failed listing runbooks, reason=%v", err)})
+			return
+		}
+		runbookList.Repositories = append(runbookList.Repositories, *repositoryList)
+	}
+
+	c.JSON(http.StatusOK, runbookList)
+}
+
+// GetRunbookConfiguration
+//
+//	@Summary		Get Runbook Configuration
+//	@Description	Get Runbook Configuration
+//	@Tags			Runbooks
+//	@Accept			json
+//	@Produce		json
+//	@Success		200			{object}	openapi.RunbookConfiguration
+//	@Failure		400,404,422,500	{object}	openapi.HTTPError
+//	@Router			/runbooks/configurations [get]
+func GetRunbookConfiguration(c *gin.Context) {
+	ctx := storagev2.ParseContext(c)
+
+	runbooks, err := models.GetRunbookConfigurationByOrgID(models.DB, ctx.GetOrgID())
+	if err != nil {
+		if errors.Is(err, models.ErrNotFound) {
+			c.JSON(http.StatusNotFound, gin.H{"message": "runbook configuration not found"})
+			return
+		}
+
+		log.Infof("failed fetching runbook configuration, err=%v", err)
+		c.JSON(http.StatusUnprocessableEntity, gin.H{"message": "failed fetching runbook configuration"})
```

Comment:
If it's not a not found error, it's an internal error. Make sure to return the message to the client as well